src/control/cmd/daos_agent/infocache_test.go

```diff
@@ -132,6 +132,58 @@ func TestInfoCacheInit(t *testing.T) {
 	}
 }
 
+func TestInfoCacheInitWithDeviceFiltering(t *testing.T) {
+	const (
+		ETHER      = 1
+		INFINIBAND = 32
+	)
+
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+	enabled := atm.NewBool(true)
+	scanResults := []netdetect.FabricScan{
+		{Provider: "ofi+sockets", DeviceName: "eth0_node0", NUMANode: 0, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "eth1_node0", NUMANode: 0, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "eth2_node0", NUMANode: 0, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "ib0_node0", NUMANode: 0, NetDevClass: INFINIBAND},
+		{Provider: "ofi+sockets", DeviceName: "eth0_node1", NUMANode: 1, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "ib1_node1", NUMANode: 1, NetDevClass: INFINIBAND},
+		{Provider: "ofi+sockets", DeviceName: "eth0_node2", NUMANode: 2, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "eth1_node2", NUMANode: 2, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "eth2_node2", NUMANode: 2, NetDevClass: ETHER},
+		{Provider: "ofi+sockets", DeviceName: "eth3_node2", NUMANode: 2, NetDevClass: ETHER}}
+
+	aiCache := attachInfoCache{log: log, enabled: enabled}
+	resp := &mgmtpb.GetAttachInfoResp{NetDevClass: ETHER}
+	err := aiCache.initResponseCache(resp, scanResults)
+	common.AssertEqual(t, err, nil, "initResponseCache error")
```

Comment:
Wondering if this should go inside the loop. Then you could also test the infiniband case here.

Reply:
Nice idea.  Done.